### PR TITLE
KAFKA-13963: Clarified java doc for processors api

### PR DESCRIPTION
**Pull request by tkaszuba**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [x] Verify design and implementation 
- [x] Verify test coverage and CI build status
- [x] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by Kvicii — APPROVED** [review]

File: streams/src/main/java/org/apache/kafka/streams/TopologyDescription.java [hunk 1]
```diff
@@ -30,6 +30,7 @@
  * In contrast, two sub-topologies are not connected but can be linked to each other via topics, i.e., if one
  * sub-topology {@link Topology#addSink(String, String, String...) writes} into a topic and another sub-topology
  * {@link Topology#addSource(String, String...) reads} from the same topic.
+ * Processors and Transformers created with the Processor API are treated as black boxes and are not represented in the topology graph.
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/TopologyDescription.java]: > are not represented

Sounds a little bit like "missing". Also, using the (plain) Processor API, there are only `Processors` anyway -- `Transformers` are part of the DSL-PAPI-integration.

If you use the Processor API, you would still connect Processors to define your graph (eg, `addProcessor(..., <parentProcessor>)`  and those connections are represented in the `TopologyDescription`.

The point from the jira was (if I understood it correctly), that `context.forward()` (which does not define the actually structure of the topology, as it does not connect `Processor` but just uses _existing_ connections) is not part of the `TopologyDescription`.

Comment (tkaszuba) [reply streams/src/main/java/org/apache/kafka/streams/TopologyDescription.java]: that is correct, the issue is with context.forward

Comment (mjsax) [reply streams/src/main/java/org/apache/kafka/streams/TopologyDescription.java]: Can you update the PR to re-phrase it so it's easier to understand?

Comment (tkaszuba) [reply streams/src/main/java/org/apache/kafka/streams/TopologyDescription.java]: ok, updated

**Comment by mjsax** [comment]
Thanks for the PR! Merged to `trunk`.

**Pull request merged**